core/src/main/java/fr/sncf/osrd/envelope/EnvelopeBuilder.java

```diff
@@ -0,0 +1,247 @@
+package fr.sncf.osrd.envelope;
+
+import static fr.sncf.osrd.envelope.EnvelopeBuilder.EnvelopeChangeType.*;
+
+import java.util.ArrayList;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.util.Collections;
+
+public final class EnvelopeBuilder {
```

Comment:
I don't understand everything that's going on in this class, but I really like its public interface